server/service/src/processors/assign_requisition_number/assign_requisition_number.rs

```diff
@@ -0,0 +1,84 @@
+use repository::{
+    ChangelogFilter, ChangelogRow, ChangelogTableName, EqualFilter, KeyType, NumberRowType,
+    RequisitionRow, RequisitionRowRepository, RequisitionType,
+};
+
+use crate::{
+    number::next_number,
+    processors::general_processor::{Processor, ProcessorError},
+    service_provider::{ServiceContext, ServiceProvider},
+    sync::ActiveStoresOnSite,
+};
+
+const DESCRIPTION: &str = "Assign requisition number to a response requisition";
+
+pub(crate) struct AssignRequisitionNumber;
+
+impl Processor for AssignRequisitionNumber {
+    fn get_description(&self) -> String {
+        DESCRIPTION.to_string()
+    }
+
+    fn try_process_record(
+        &self,
+        ctx: &ServiceContext,
+        _service_provider: &ServiceProvider,
+        changelog: &ChangelogRow,
+    ) -> Result<Option<String>, ProcessorError> {
+        let repo = RequisitionRowRepository::new(&ctx.connection);
+
+        let requisition =
+            repo.find_one_by_id(&changelog.record_id)?
+                .ok_or(ProcessorError::RecordNotFound(
+                    "Requisition".to_string(),
+                    changelog.record_id.clone(),
+                ))?;
+
+        // Only assign requisition number to response requisitions
```

Comment:
Seriously having the comments where they're relevant rather than the big block at the top is much easier reading